WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -2299,26 +2278,22 @@ public void onMediaChanged(OnMediaChanged event) {
     /**
      * Starts the upload service to upload selected media.
      */
-    private void startUploadService() {
-        if (mPendingUploads != null && !mPendingUploads.isEmpty()) {
-            final ArrayList<MediaModel> mediaList = new ArrayList<>();
-            for (MediaModel media : mPendingUploads) {
-                if (MediaUploadState.QUEUED.toString().equals(media.getUploadState())) {
-                    mediaList.add(media);
-                }
-            }
+    private void startUploadService(MediaModel media) {
+        final ArrayList<MediaModel> mediaList = new ArrayList<>();
+        if (MediaUploadState.QUEUED.toString().equals(media.getUploadState())) {
```

Comment:
I think this method can be simplified by doing a `return` if the media is already queued. So:

```
private void startUploadService(MediaModel media) {
    if (MediaUploadState.QUEUED.toString().equals(media.getUploadState())) {
        return;
    }

    final ArrayList<MediaModel> mediaList = new ArrayList<>();
    mediaList.add(media);

    // before starting the service, we need to update the posts' contents so we are sure the service
    // can retrieve it from there on
    savePostAsync(new AfterSavePostListener() {
        @Override
        public void onPostSave() {
            UploadService.uploadMedia(EditPostActivity.this, mediaList);
        }
    });
}
```

Reply:
Just for info for the casual reader of this PR, the actual meaning of the code in there is that the media item needs to have the `QUEUED` state in order to be passed to the `UploadService`. I could yet use the simplifiation by adding a negation operator at the start of that if, like this:
```
if (!MediaUploadState.QUEUED.toString().equals(media.getUploadState())) {
        return;
    }
```
would that make sense @nbradbury ?

Reply:
Oh, right - I completely reversed that. Yes, that makes sense, and perhaps add a comment explaining it.

Reply:
addressed in f911bed